backend/src/api/user/user.api.ts

```diff
@@ -1,54 +1,64 @@
 import { Router } from 'express';
-import { ApiPath, HttpCode, UsersApiPath } from '~/common/enums';
+import { ApiPath, HttpCode, UsersApiPath, UserType } from '~/common/enums';
 import { userService } from '~/services/services';
 
 const initUserApi = (apiRouter: Router): Router => {
   const userRouter = Router();
 
   apiRouter.use(ApiPath.USERS, userRouter);
 
-  userRouter.get(UsersApiPath.ROOT, async (_req, res) => {
+  userRouter.get(UsersApiPath.ROOT, async (_req, res, next) => {
     try {
-      const users = await userService.getAllUsers()
+      const users = await userService.getAllUsers();
       res.status(HttpCode.OK).json(users);
     } catch(error) {
-      res.status(HttpCode.NOT_FOUND).json(error);
+      next(error);
     }
   });
 
-  userRouter.get(UsersApiPath.$ID, async (_req, res) => {
+  userRouter.get(UsersApiPath.$TYPE, async (_req, res, next) => {
+    try {
+      const users = await userService.getUsersByType(_req.params.type as UserType);
```

Comment:
```suggestion
  userRouter.get(UsersApiPath.$TYPE, async (req, res, next) => {
    try {
      const users = await userService.getUsersByType(req.params.type as UserType);
```
The underscore uses for the non-using params